### PR TITLE
Fix performance for CUDA 9.2 / 10.0

### PR DESCRIPTION
**Pull request by lukasm91**
- make constexpr compiler dependent whenever possible (replace with macro `GT_CONSTEXPR`)
- remove `const_expr::move` and `const_expr::forward`
- add `wstd::move` and `wstd::forward` (move and forward are only constexpr if compiler does not suffer from it)

### CODE REVIEW

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do peftest

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
do perftest

File: include/gridtools/common/generic_metafunctions/utility.hpp [hunk 1]
```diff
@@ -10,33 +10,32 @@
...
 #include <type_traits>
 #include <utility>
 
+#include "../defs.hpp"
+#include "../host_device.hpp"
+
 namespace gridtools {
     /**
-     *  `std::forward`/`std::move` versions that are guarantied to be constexpr
+     *  `std::forward`/`std::move` versions that are guaranteed to be not constexpr
```
Comment (anstaf) [inline include/gridtools/common/generic_metafunctions/utility.hpp]: I would explain the rationale for introducing `wstd` in more details

File: include/gridtools/meta/fold.hpp [hunk 2]
```diff
@@ -55,7 +55,8 @@ namespace gridtools {
             };
             template <template <class...> class F,
                 class S,
-                template <class...> class L,
+                template <class...>
```
Comment (anstaf) [inline include/gridtools/meta/fold.hpp]: that's strange. it looks like your `clang-format` behaves  differently then mine. mine has the version `6.0.0`

Comment (lukasm91) [reply include/gridtools/meta/fold.hpp]: That's strange yes... I have `8.0.0`. I reverted these changes and check this in more detail later.

**Comment by lukasm91** [comment]
launch strgrid

**Comment by lukasm91** [comment]
launch icgrid

**Comment by lukasm91** [comment]
do perftest

**Comment by lukasm91** [comment]
do perftest

**Pull request merged**